File: src/lib/platform/updater.ts
import { invokeTauriCommand } from '$lib/platform/tauri'

export interface SourceUpdateStatus {
  currentBranch?: string | null
  dirty: boolean
  headCommit?: string | null
  installPath: string
  mainCommit?: string | null
  sourceDir: string
  sourceExists: boolean
  updateAvailable: boolean
}

export interface SourceUpdateStepResult {
  label: string
  ok: boolean
  output: string
}

export interface SourceUpdateResult {
  installPath: string
  sourceDir: string
  steps: SourceUpdateStepResult[]
  updated: boolean
}

export function checkSourceUpdate(): Promise<SourceUpdateStatus> {
  return invokeTauriCommand<SourceUpdateStatus>('check_source_update')
}

export function runSourceUpdate(): Promise<SourceUpdateResult> {
  return invokeTauriCommand<SourceUpdateResult>('run_source_update')
}

File: src-tauri/src/commands/updater.rs
use crate::{errors::AppResult, updater};

#[tauri::command]
pub async fn check_source_update() -> AppResult<updater::SourceUpdateStatus> {
    updater::check_source_update()
}

#[tauri::command]
pub async fn run_source_update() -> AppResult<updater::SourceUpdateResult> {
    updater::run_source_update()
}
